pkg/debug/diff.go

```diff
@@ -0,0 +1,106 @@
+package debug
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sort"
+	"strings"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/kylelemons/godebug/diff"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// GenerateDiff attempts to pull a dependency list from a previous job (job, jobID) and generate a diff against a provided string
+func GenerateDiff(jobName, jobID, phase, dependencies string) (string, error) {
+	resp, err := http.Get(fmt.Sprintf("https://storage.googleapis.com/origin-ci-test/logs/%s/%s/artifacts/%s/dependencies.txt", jobName, jobID, phase))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+
+	return diff.Diff(string(body), dependencies), nil
+}
+
+// GenerateDependencies creates a list of images and the MCC hash
+func GenerateDependencies(kube kubernetes.Interface) (dependencies string, err error) {
+	pods, err := kube.CoreV1().Pods("").List(metav1.ListOptions{})
```

Comment:
Probably need to filter out osde2e specific things from this list -- maybe we should do this at cluster creation rather than cleanup?

Reply:
Disagree here. We, and our tests, shouldn't be exempt from this list. If, for example, a redis image we're using in a workload test changes, we didn't know, and that test starts failing, we'd immediately know the image changed and is causing the issue.

Reply:
Hrrrm, hadn't thought about that. Good point!